core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingTransformer.java

```diff
@@ -232,14 +267,23 @@ private class IndividualTransformer implements AsyncResponseTransformer<Response
         public CompletableFuture<ResponseT> prepare() {
             this.individualFuture = new CompletableFuture<>();
             if (preparedCalled.compareAndSet(false, true)) {
+                if (isCancelled.get()) {
+                    return individualFuture;
+                }
                 CompletableFuture<ResultT> upstreamFuture = upstreamResponseTransformer.prepare();
                 if (!resultFuture.isDone()) {
                     CompletableFutureUtils.forwardResultTo(upstreamFuture, resultFuture);
                 }
             }
+            resultFuture.whenComplete((r, e) -> {
+                if (e == null) {
+                    return;
+                }
+                individualFuture.completeExceptionally(e);
+            });
             individualFuture.whenComplete((r, e) -> {
                 if (isCancelled.get()) {
-                    handleCancelState();
+                    handleSubscriptionCancel();
```

Comment:
Can you remind me why we need to check isCancelled here for every individual future?